release.rb

```diff
@@ -0,0 +1,215 @@
+       die("Failed to tag version. Aborting.") unless tagged_version
+
+       pushed_tag = system("git push origin #{version_tag}")
+       die("Failed to push tag #{version_tag} to origin. Aborting.") unless pushed_tag
+    end
+
+    if set_podspec_version
+      `rm #{release_notes}`
+      pushed_to_origin = system("git push origin #{remote_branch}")
+      die("Failed to push to origin. Aborting.") unless pushed_to_origin
+
+      puts " > Pushed version to origin"
+    end
+
+    puts
+    puts "---------------- Released as $VERSION_TAG ----------------"  
+    puts
+    puts "Archiving carthage release..."
+    
+    archived_carthage_release = system("#{carthage} archive #{carthage_framework_name}")
+    die("Failed to archive framework for carthage. Aborting.") unless archived_carthage_release
+
+    puts
+    #puts "Pushing to pod trunk..."
+
+    #`#{pod} trunk push #{pod_spec}`
+
+    puts
+    puts "================ Finalizing the Release ================"
+    puts
+    puts " - Go to #{github_tags_url} and mark this as a release."
+    puts "   - Paste the contents of #{release_notes} into the release notes. Tweak for Github styling."
+    puts "   - Attach #{carthage_framework_name}.framework.zip to it."
+    puts " - Announce!"
+
+    `rm #{pod_spec}.backup`
+  end
+
+  def self.die(error_message)
+    raise("[ERROR]: #{error_message}")
+  end
+
+  def self.verify_availability(command) 
+    system("which #{command}")
+  end
+
+  def self.verify_version(version, force) 
+    die("This tag (#{version}) is an incorrect format. You should remove the 'v' prefix.") unless /^[vv]/.match(version)
+    die("This tag (#{version}) is an incorrect format. It should be in 'v{MAJOR}.{MINOR}.{PATCH}(-{PRERELEASE_NAME}.{PRERELEASE_VERSION})' form.") unless /^v(\d+\.)(\d+\.)(\d+)(-\w+(\.\d+)?)?$/.match(version)
+ 
+    puts " > Is this version (#{version}) unique?"
+
+    tag_exists = `git describe --exact-match #{version} > /dev/null 2>&1`
+    if tag_exists then
+      if force then
+        puts " > NO, but force was specified."
+      else
+        die("This tag (#{version}) already exists. Aborting. Use the force option to override.")
+      end
+    else
+      puts " > YES, tag is unique."
+    end
+  end
+  
+  def self.gather_release_notes(release_notes_file_path, version, pod_name) 
+    unless !release_notes_file_path.nil? && File.file?(release_notes_file_path)
+      puts " > Failed to find #{release_notes_file_path}. Prompting editor"
+      if ENV["EDITOR"].nil?
+        die("No default editor set. Assign an editor to the $EDITOR environment var. Aborting...")
+      end
+      release_notes_file_path = '.release-changes'
+      latest_tag = `git for-each-ref refs/tags --sort=-refname --format="%(refname:short)" | grep -E "^v\\d+\\.\\d+\\.\\d+(-\\w+(\\.\\d)?)?\\$" | ruby -e 'puts STDIN.read.split("\n").sort { |a,b| Gem::Version.new(a.gsub(/^v/, "")) <=> Gem::Version.new(b.gsub(/^v/, "")) }.last'`
```

Comment:
This line is ludicrously long -- I'll address it in the next iteration. I forgot to do so in this one :sweat_smile: